jcl/src/java.management/share/classes/java/lang/management/ThreadInfo.java

```diff
@@ -121,119 +89,55 @@ private ThreadInfo(long threadIdVal, String threadNameVal,
-		this.stackTraces = stackTraceVal;
-		this.lockInfo = lockInfo;
-		this.lockedMonitors = lockedMonitors;
-		this.lockedSynchronizers = lockedSynchronizers;
-		/*[IF Sidecar19-SE]*/
-		this.daemon = daemon;
-		this.priority = priority;
-		/*[ENDIF]*/
+		MonitorInfoBase[] monInfoBases = null;
+		if (null != lockedMonitors) {
+			monInfoBases = Arrays.stream(lockedMonitors)
+					.map(m -> m.getBaseInfo())
+					.collect(Collectors.toList())
+					.toArray(EMPTY_MONINFO_BASES);
+		}
+		
+		LockInfoBase[] lockedSyncs = null;
+		if (null != lockedSynchronizers) {
+			lockedSyncs = Arrays.stream(lockedSynchronizers)
+					.map(m -> m.getBaseInfo())
+					.collect(Collectors.toList())
+					.toArray(EMPTY_LOCKINFOBASE_ARRAY);
+		}
+		LockInfoBase lckInfo = null;
+		if (null != lockInfo) {
+			lckInfo = lockInfo.getBaseInfo();
+		}
+		baseInfo = new ThreadInfoBase(threadIdVal, threadNameVal, threadStateVal, suspendedVal, inNativeVal,
+				blockedCountVal, blockedTimeVal, waitedCountVal, waitedTimeVal, lockNameVal, lockOwnerIdVal,
+				lockOwnerNameVal, stackTraceVal, lckInfo, monInfoBases, lockedSyncs
+/*[IF Sidecar19-SE]*/
+				, daemon, priority
+/*[ENDIF]*/
+				);
 	}
 
 	/**
-	 * @param thread
-	 * @param state
-	 * @param isSuspended
-	 * @param isInNative
-	 * @param blockedCount
-	 * @param blockedTime
-	 * @param waitedCount
-	 * @param waitedTime
-	 * @param stackTrace
-	 * @param blockingMonitor
-	 * @param lockOwner
+	 * Constructor for use by subclasses
+	 * @param base data for the object
 	 */
-	private ThreadInfo(Thread thread, long nativeId, int state, boolean isSuspended,
-			boolean isInNative, long blockedCount, long blockedTime,
-			long waitedCount, long waitedTime, StackTraceElement[] stackTrace,
-			Object blockingMonitor, Thread lockOwner) {
-		this.threadId = thread.getId();
-		this.nativeTId = nativeId;
-		this.threadName = thread.getName();
-		/*[IF Sidecar19-SE]*/
-		this.daemon = thread.isDaemon();
-		this.priority = thread.getPriority();
-		/*[ENDIF]*/
-
-		// Use the supplied state int to index into the array returned
-		// by values() which should be all thread states in the order
-		// they have been declared. Doing this rather than just issuing a
-		// call to thread.getState() because the Thread state may have
-		// changed after the VM issued the call to this method.
-		this.threadState = Thread.State.values()[state];
-		this.suspended = isSuspended;
-		this.inNative = isInNative;
-		this.blockedCount = blockedCount;
-		this.blockedTime = blockedTime;
-		this.waitedCount = waitedCount;
-		this.waitedTime = waitedTime;
-
-		if (lockOwner != null) {
-			this.lockOwnerId = lockOwner.getId();
-			this.lockOwnerName = lockOwner.getName();
-		}
-
-		this.stackTraces = stackTrace;
-		if (blockingMonitor != null) {
-			this.lockName = blockingMonitor.getClass().getName()
-					+ '@'
-					+ Integer.toHexString(System
-							.identityHashCode(blockingMonitor));
-		}
+	protected ThreadInfo(ThreadInfoBase base) {
```

Comment:
protected methods/constructors cannot be added to classes which are part of the api specification, it changes the api. Please fix them all.

Reply:
Done.  The constructor is now called using a method handle obtained by reflection.